client/verta/verta/_internal_utils/model_dependencies.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+
+import ast
+import inspect
+import textwrap
+from types import ModuleType
+from typing import Callable, get_type_hints, List, Tuple, Type
+
+from verta.registry import VertaModelBase
+
+
+def list_class_functions(model_class: Type[VertaModelBase]) -> List[Tuple[str, Callable]]:
+    """List all the functions present in the provided class object."""
+    return inspect.getmembers(model_class, predicate=inspect.isfunction)
+
+
+def unwrap(func: Callable) -> Callable:
+    """Unwraps a function to get the original function object in a non-blocking manner."""
+    try:
+        return inspect.unwrap(func)  # returns last object in chain
+    except ValueError:  # if a cycle is encountered
+        return func
+
+
+def list_modules_in_function_definition(func: Callable) -> List[ModuleType]:
+    """List all modules called within the body of the provided function."""
+    _func = unwrap(func)
+    return [
+        value for key, value in inspect.getclosurevars(_func).globals.items()
+        if isinstance(value, ModuleType)
+    ]
+
+
+def list_modules_in_function_signature(func: Callable) -> List[ModuleType]:
+    """List all modules used in type hints in the provided function's arguments
+    and return type hint."""
+    _func = unwrap(func)
+    hints = get_type_hints(_func)
+    arg_hints = { k: v for k, v in hints.items() if k != 'return' }
+    return_hint = { k: v for k, v in hints.items() if k == 'return' }
+
+    modules = [ inspect.getmodule(value) for key, value in arg_hints.items() ]
+
+    if return_hint:
+        mod = inspect.getmodule(return_hint['return'])
+        if mod.__name__ == 'typing':
+            ret_ann = inspect.signature(_func).return_annotation
+            nested_args = ret_ann.__args__
+            for a in nested_args:
```

Comment:
This probably doesn't handle deeper nesting, right? Like `Dict[str, List[np.array]]`

Reply:
Likely not.  We should be able to do this with a recursive function of some kind though right?  I'll add it to the "would be nice to fix if we have time to spare at the end of next sprint" pile.

Reply:
Same here. Adding to the list